### PR TITLE
Runtime configurable server read/write timeouts

### PR DESCRIPTION
**Pull request by gmalouf**
## Summary

Made configurations for read/write timeouts configurable at runtime, with sensible defaults and a basic generation formula for handler timeout.

## Test Plan

Started up server locally, exercised both using default values and passing in command line overrides for each.

### CODE REVIEW

**Review by winder — APPROVED** [review]

**Comment by bricerisingalgorand** [comment]
Do we provide these timeouts in seconds or milliseconds?

**Comment by gmalouf** [comment]
// ParseDuration parses a duration string.
// A duration string is a possibly signed sequence of
// decimal numbers, each with optional fraction and a unit suffix,
// such as "300ms", "-1.5h" or "2h45m".
// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
func ParseDuration(s string) (Duration, error) {

So you can do "5s" or "5000ms" to get the same parsed duration value.

File: api/server.go [hunk 1]
```diff
@@ -39,6 +30,23 @@ type ExtraOptions struct {
...
+	//off of this.
+	WriteTimeout time.Duration
+
+	// ReadTimeout is the maximum duration for reading the entire request, including the body.
+	ReadTimeout time.Duration
+}
+
+func (e ExtraOptions) handlerTimeout() time.Duration {
+	// Basically, if write timeout is 2 seconds or greater, subtract a second.
+	// If less, subtract 10% as a safety valve.
```
Comment (tsachiherman) [inline api/server.go:44]: valve -> value

Comment (gmalouf) [reply api/server.go:44]: Funny, I was definitely intending safety valve as a phrase here.

**Pull request merged**